rusty_pizza_server/src/order_model/meal.rs

```diff
@@ -42,6 +101,132 @@ impl<'a> Iterator for SpecialsMut<'a> {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub struct MealBuilder {
```

Comment:
I'm not particularly happy with the design of the builder:
- imo mandatory parameters `meal_id` and `price` are not enforced to be set and default to `""` which is not helpful when you try to order something
- specials will soon have IDs, making the use of an owned `SpecialFactory` an actual pain
- needing to pass the `MealFactory` to the "`build`" method seems complicated, I would expect it to have no parameters

How about changing this the following way:
- `MealFactory` creates `MealBuilder` instead of `Meal` and seeds it with an (immutable) ID
- `MealFactory` also takes the two mandatory parameters and passes them (privately) to the builder
- `MealFactory` also passes a lifetime bound reference to its own `SpecialFactory` (the lifetime constraint is important to make this work!) to the builder
- `MealBuilder` creates the `Meal` in `meal`

What's your opinion on this?